model/shared/src/main/scala/explore/model/ConstraintsSummary.scala

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.model
+
+import cats._
+import eu.timepit.refined.types.string.NonEmptyString
+import io.circe.Decoder
+import io.circe.Encoder
+import io.circe.generic.semiauto._
+import io.circe.refined._
+import lucuma.core.enum.CloudExtinction
+import lucuma.core.enum.ImageQuality
+import lucuma.core.enum.SkyBackground
+import lucuma.core.enum.WaterVapor
+import lucuma.core.model.ConstraintSet
+import monocle.macros.Lenses
+
+@Lenses
+final case class ConstraintsSummary(
+  name:            NonEmptyString,
+  id:              ConstraintSet.Id,
+  imageQuality:    ImageQuality,
+  cloudExtinction: CloudExtinction,
+  skyBackground:   SkyBackground,
+  waterVapor:      WaterVapor
+) {
+  def summaryString: String =
+    s"${imageQuality.label} ${cloudExtinction.label} ${skyBackground.label} ${waterVapor.label}"
+}
+
+object ConstraintsSummary {
+
+  implicit val constraintSummaryDecoder: Decoder[ConstraintsSummary] = deriveDecoder
+  implicit val constraintSummaryEncoder: Encoder[ConstraintsSummary] = deriveEncoder
+
+  implicit val eqConstraintSummary: Eq[ConstraintsSummary] = Eq.fromUniversalEquals
```

Comment:
We may want to make these from other `Eq` instances doing e.g. `Eq.by` unless all members have trivial `Eq` instances

Reply:
Switched to `Eq.by`. I hope Eq and Reusability support the new deriving in Scala 3. I don't like how they can get out of date without warning. I found and instance of that for `Reusabilty.by` recently.